korman/exporter/mesh.py

```diff
@@ -214,6 +218,24 @@ def _export_geometry(self, bo, mesh, materials, geospans):
                                    ((src.color3[0] + src.color3[1] + src.color3[2]) / 3),
                                    ((src.color4[0] + src.color4[1] + src.color4[2]) / 3))
 
+            if has_bumpmap is not None:
+                gradPass = []
+                gradUVWs = []
+
+                if len(tessface.vertices) != 3:
+                    gradPass.append([tessface.vertices[0], tessface.vertices[1], tessface.vertices[2]])
+                    gradPass.append([tessface.vertices[0], tessface.vertices[2], tessface.vertices[3]])
+                    gradUVWs.append([tuple([uvw[0] for uvw in tessface_uvws]), tuple([uvw[1] for uvw in tessface_uvws]), tuple([uvw[2] for uvw in tessface_uvws])])
+                    gradUVWs.append([tuple([uvw[0] for uvw in tessface_uvws]), tuple([uvw[2] for uvw in tessface_uvws]), tuple([uvw[3] for uvw in tessface_uvws])])
+                else:
+                    gradPass.append(tessface.vertices)
+                    gradUVWs.append([tuple([uvw[0] for uvw in tessface_uvws]), tuple([uvw[1] for uvw in tessface_uvws]), tuple([uvw[2] for uvw in tessface_uvws])])
```

Comment:
I mean change your list comprehensions to generator comprehensions. Change the brackets to parens. That paradigm is both more logical and efficient here. https://www.python.org/dev/peps/pep-0289/